float32 p_out
float32 v_out
float32 a_out